Sources/tasks/CreateAttachment-Notify.php

```diff
@@ -0,0 +1,160 @@
+<?php
+/**
+ * This file contains background notification code for moderators
+ * to approve attachments.
+ *
+ * Simple Machines Forum (SMF)
+ *
+ * @package SMF
+ * @author Simple Machines http://www.simplemachines.org
+ * @copyright 2018 Simple Machines and individual contributors
+ * @license http://www.simplemachines.org/about/smf/license.php BSD
+ *
+ * @version 2.1 Beta 4
+ */
+
+/**
+ * Class CreateAttachment_Notify_Background
+ */
+class CreateAttachment_Notify_Background extends SMF_BackgroundTask
+{
+	/**
+	 * This executes the task - loads up the info, puts the email in the queue and inserts any alerts as needed.
+	 * @return bool Always returns true
+	 */
+	public function execute()
+	{
+		global $smcFunc, $sourcedir, $scripturl, $modSettings, $language;
+
+		// Validate the attachment does exist and is the right approval state.
+		$request = $smcFunc['db_query']('', '
+			SELECT a.id_attach, m.id_board, m.id_msg, m.id_topic, m.id_member, m.subject
+			FROM {db_prefix}attachments AS a
+				INNER JOIN {db_prefix}messages AS m ON (m.id_msg = a.id_msg)
```

Comment:
Like always and like the old query,
i keep recommand to drop the word inner here.

Reply:
Why? Can you explain, please?

Reply:
This is how queries are done everywhere in SMF.

Perhaps this is something to look into once the next major version of SMF starts being developed.